src/modules/networking/src/lib/Networking.cpp

```diff
@@ -391,53 +402,84 @@ void NetworkingObjectBase::GenerateInterfaceTypesMap()
+                        std::getline(interfaceDataStream, token, ' ');
+                    }
+                    while(token.empty());
+
+                    std::string interfaceType = token;        
+                    if ((!interfaceName.empty()) && (!interfaceType.empty()))
                     {
-                        interfaceName = data;
-                        do
+                        std::map<std::string, std::string>::iterator interfaceTypesMapIterator = this->m_interfaceTypesMap.find(interfaceName);
+                        if (interfaceTypesMapIterator != this->m_interfaceTypesMap.end())
                         {
-                            std::getline(interfaceDataStream, data, ' ');
+                            interfaceTypesMapIterator->second = interfaceType;
                         }
-                        while(data.empty());
-
-                        interfaceType = data;          
-                        if ((!interfaceName.empty()) && (!interfaceType.empty()))
+                        else
                         {
-                            interfaceTypesMapIterator = this->m_interfaceTypesMap.find(interfaceName);
-                            if (interfaceTypesMapIterator != this->m_interfaceTypesMap.end())
-                            {
-                                interfaceTypesMapIterator->second = interfaceType;
-                            }
-                            else
-                            {
-                                this->m_interfaceTypesMap.insert(std::pair<std::string, std::string>(interfaceName, interfaceType));
-                            }
+                            this->m_interfaceTypesMap.insert(std::pair<std::string, std::string>(interfaceName, interfaceType));
                         }
                     }
                 }
             }
+        }
+    }
+
+    if ((this->m_interfaceTypesMap.empty()) && (this->m_networkManagementService == NetworkManagementService::SystemdNetworkd))
+    {
+        this->m_networkManagementService = NetworkManagementService::Unknown;
+    }
+}
 
-            interfaceName.clear();
-            interfaceType.clear();
+void NetworkingObjectBase::GenerateInterfaceTypesMap()
+{
+    this->m_interfaceTypesMap.clear();
+
+    if ((this->m_networkManagementService == NetworkManagementService::NetworkManager) || (this->m_networkManagementService == NetworkManagementService::Unknown))
+    {
+        GetInterfaceTypesFromNetworkManager();
+    }
+    if ((this->m_networkManagementService == NetworkManagementService::SystemdNetworkd) || (this->m_networkManagementService == NetworkManagementService::Unknown))
+    {
+        GetInterfaceTypesFromSystemdNetworkd();
+    }
+    if (this->m_networkManagementService == NetworkManagementService::Unknown)
+    {
+        if (IsFullLoggingEnabled())
+        {
+            OsConfigLogError(NetworkingLog::Get(), "Network interface management service not found");
         }
     }

```

Comment:
Should lines 474 and 478 be 'else if' rather than 'if'?

Reply:
Each should be an 'if' statement. This is where Networking discovers the network management service enabled on the system (Network Manager or systemd-networkd) and caches the result in m_networkManagementService. Code contained in the first 'if' checks if Network Manager is enabled and collects InterfaceTypes data if possible. If Network Manager is not enabled, code contained in the second 'if' checks if systemd-networkd is enabled and collects InterfaceTypes data if possible. If neither Network Manager nor systemd-networkd are enabled, and full logging is enabled, code contained in the third 'if' logs a Networking error. It is important that this flow is implemented as a series of 'if' statements because in cases where one or more target network management services are not found, the module needs to continue trying new queries, or log an error if none of the known services are discovered.